examples/webapp/webapp.py

```diff
@@ -0,0 +1,202 @@
+
+        script = self._ctx.new_script(timeout=timedelta(seconds=30))
+        script.run("/bin/run_rqlite.sh")
+        yield script
+
+    async def reset(self):
+        # We don't have to do anything when the service is restarted
+        pass
+
+
+async def main(subnet_tag, payment_driver, payment_network, port):
+    async with Golem(
+        budget=1.0,
+        subnet_tag=subnet_tag,
+        payment_driver=payment_driver,
+        payment_network=payment_network,
+    ) as golem:
+        print_env_info(golem)
+
+        network = await golem.create_network("192.168.0.1/24")
+        async with network:
+            db_cluster = await golem.run_service(DbService, network=network)
+            db_instance = db_cluster.instances[0]
+
+            while db_instance.state != ServiceState.running:
+                await asyncio.sleep(5)
+                print(db_instance)
+
+            print(
+                f"{TEXT_COLOR_CYAN}DB instance started, spawning the web server{TEXT_COLOR_DEFAULT}"
+            )
+
+            commissioning_time = datetime.now()
+
+            web_cluster = await golem.run_service(
+                HttpService,
+                network=network,
+                instance_params=[{"db_address": db_instance.network_node.ip}],
+            )
+
+            instances = web_cluster.instances
+
+            def still_starting():
+                return any(
+                    i.state in (ServiceState.pending, ServiceState.starting) for i in instances
+                )
+
+            # wait until all remote http instances are started
+
+            while still_starting() and datetime.now() < commissioning_time + STARTING_TIMEOUT:
+                print(instances)
+                await asyncio.sleep(5)
+
+            if still_starting():
+                raise Exception(
+                    f"Failed to start web instances after {STARTING_TIMEOUT.total_seconds()} seconds"
+                )
+
+            # service instances started, start the local HTTP server
+
+            proxy = LocalHttpProxy(web_cluster, port)
+            await proxy.run()
+
+            print(
+                f"{TEXT_COLOR_CYAN}Local HTTP server listening on:\nhttp://localhost:{port}{TEXT_COLOR_DEFAULT}"
+            )
+
+            # wait until Ctrl-C
+
+            while True:
+                print(instances)
+                try:
+                    await asyncio.sleep(10)
+                except (KeyboardInterrupt, asyncio.CancelledError):
+                    break
+
+            # perform the shutdown of the local http server and the service cluster
+
+            await proxy.stop()
+            print(f"{TEXT_COLOR_CYAN}HTTP server stopped{TEXT_COLOR_DEFAULT}")
+
+            web_cluster.stop()
+            db_cluster.stop()
+
+            cnt = 0
+            while cnt < 3 and any(
+                s.is_available for s in web_cluster.instances + db_cluster.instances
+            ):
+                print(instances)
```

Comment:
`instances` here are only `web_cluster` instances, but we'll be printing them also with only `db_cluster` instance running. This might be confusing.

Reply:
I don't understand this comment

Reply:
aaah, okay, I see my error now :)